### PR TITLE
Add entity interaction event.

### PR DESCRIPTION
**Pull request by amberkowalski**
Adds an entity interaction event that gets fired when the server gets an entity interact packet. 

This is my first PR on the refactor so let me know if I put code in the wrong places or did anything wrong please.

TODO(?)
- [ ] The server is *supposed* to verify that the interaction is "valid" before it actually does anything. 
- [x] Sprinkle in some docs
- [x] Test this properly when entities can be spawned ~~(I can't figure out how to, there might already be a method of doing so)~~

### CODE REVIEW

**Comment by amberkowalski** [comment]
Just did a little testing, it appears to work.

**Review by caelunshun — APPROVED** [review]
LGTM, thanks!

As for validation—we should probably wait for a physics implementation (#357) so we have the necessary bounding box methods.

**Comment by amberkowalski** [comment]
mmmmm that's a mess. Anyway I got the kick working, and it returns a system error as well.

File: crates/common/Cargo.toml [hunk 1]
```diff
@@ -19,3 +19,4 @@ quill-common = { git = "https://github.com/feather-rs/quill", rev = "d4da5b6" }
 smartstring = "0.2"
 utils = { path = "../utils", package = "feather-utils" }
 uuid = { version = "0.8", features = [ "v4" ] }
+libcraft-core = { git = "https://github.com/amberkowalski/libcraft.git", branch = "amber-entity-interact" }
```
Comment (amberkowalski) [inline crates/common/Cargo.toml:22]: This needs to be updated to use the correct repo.

File: crates/server/Cargo.toml [hunk 2]
```diff
@@ -47,6 +47,7 @@ ureq = { version = "2", features = [ "json" ] }
 utils = { path = "../utils", package = "feather-utils" }
 uuid = "0.8"
 vec-arena = "1"
+libcraft-core = { git = "https://github.com/amberkowalski/libcraft.git", branch = "amber-entity-interact" }
```
Comment (amberkowalski) [inline crates/server/Cargo.toml:50]: This also needs to be updated to use the correct repo.

**Review by caelunshun — APPROVED** [review]
Thanks!

**Pull request merged**